core.js

```diff
@@ -1139,18 +1139,32 @@ export class FileTypeParser {
 			}
 		}
 
-		if (
-			this.checkString('WEBVTT')
-			&&	(
-				// EOF
-				tokenizer.fileInfo.size === 6
-				// One of LF/CRLF, tab, or space
-				|| (['\n', '\r\n', '\t', ' '].some(whitespace => this.checkString(whitespace, {offset: 6}))))
-		) {
-			return {
-				ext: 'vtt',
-				mime: 'text/vtt',
-			};
+		if (this.checkString('WEBVTT')) {
+			await tokenizer.ignore(6);
+			let isValidWebVTT = false;
+
+			// Try-catch to handle valid edge case of "WEBVTT<EOF>"
+			try {
+				// The WebVTT standard says that the first line should be "WEBVTT" followed by a single ASCII whitespace character
+				const whitespaceToken = await tokenizer.readToken(new Token.StringType(1, 'ascii'));
```

Comment:
It does not work like this, you are reading the very from the stream.
Secondly, if you start consuming the stream, all following on test will fail, so you have to determine the file type.

Reply:
> The reason you see the following tests did not break is because this rare situation only occures on files starting with WEBVTT, but that does not make it right.

What if someone creates a file that has `WEBVTTx` as the first seven bytes? The first six bytes aren't the full signature per the spec: "An optional UTF-8 BOM, the ASCII string "WEBVTT", and finally a space, tab, line break, or the end of the file." So I don't think it's rigorous enough to just check the first six bytes as it doesn't tell us anything about the seventh byte. With my current code, a file that starts with `WEBVTT` but is _not_ followed by a whitespace character would return `undefined`, which I think is the expected behavior.

> The 7th character you are interested in has already been peeked, the EOF has already been handled.
I know you lost the info where the EOF is, but both \0 and EOF are acceptance, there is no is issue simply testing for \0 is fine.

Sorry, I must have misunderstood you before. I thought you didn't want me to test for `\0` since it's not the same as an EOF.

Reply:
> What if someone creates a file that has WEBVTTx as the first seven bytes? The first six bytes aren't the full signature per the spec: "An optional UTF-8 BOM, the ASCII string "WEBVTT", and finally a space, tab, line break, or the end of the file." So I don't think it's rigorous enough to just check the first six bytes as it doesn't tell us anything about the seventh byte. With my current code, a file that starts with WEBVTT but is not followed by a whitespace character would return undefined, which I think is the expected behavior (because that file does not comply with the signature format for valid WebVTT files).

That is indeed the best solution, but you cannot do it by reading from stream, as that irreversible consumes the stream. If you do it by reading, you break the case of "The first six bytes aren't the full signature".

> Sorry, I must have misunderstood you before. I thought you didn't want me to test for \0 since it's not the same as an EOF.

I am sorry, as I thought `\0` was defined as a formal separator. But it is not, and you are testing on `\0` as that is the way the EOF translates in `file-type`. That is why complained about the EOF fixture file, I thought it had to to have `\0` separator.

Reply:
> `WEBVTT78WEBP123XXXX`. This should be not be recognized as `WebVTT`, but it should be recognized as `WEBP`.

Oh, I see. Does this mean the `file-type` parser ignores unrecognized byte sequences at the start of a file and continues searching until it eventually finds a valid signature? I thought magic numbers had to appear at the start of a file.

> I am sorry, as I thought \0 was defined as a formal separator. But it is not, and you are testing on \0 as that is the way the EOF translates in file-type. That is why complained about the EOF fixture file, I thought it had to to have \0 separator.

All good, I think I understand what you meant now.

Reply:
>  Does this mean the file-type parser ignores unrecognized byte sequences at the start of a file and continues searching until it eventually finds a valid signature?

Exactly.

> Edit: So just to be clear, do you want me to simplify the code to just check WEBVTT, without checking the 7th byte, and remove the other fixtures?

No I agree with you to include the 7th.

This commit was very close to perfection 😉 : 53be4b412b4b32650fb66f926961c39bd896e7ea
- Rename variable `whitespace` to `char7`
- change '\r\n` into `\r`
- Using `tokenizer.fileInfo.size === 6` is tricky, it is possible to let file-size read from a stream with an unknown length. Safer to use the `0`, assuming that is the result of an "EOF". 

, and I think it is good to go.